app/src/main/java/org/mozilla/focus/browser/BrowserNavigationOverlay.kt

```diff
@@ -406,6 +411,7 @@ class BrowserNavigationOverlay @JvmOverloads constructor(
                 megaTileTryAgainButton.requestFocus()
             }
         }
+        updateOverlayForCurrentState()
```

Comment:
Think we need to call this for `refreshTilesForInsertion()` as well, for when inserting a pin into empty container

Reply:
Is there any way to pin a tile aside from clicking the icon itself?  The icon is already updated when we do that.

Reply:
Interesting, so we call `updateOverlyCurrentState()` in `navUrlInput.onFocusChangeListener {}`. So you're right, technically we are updating `pocketVideoMegaTileView.nextFocusDownId` when we navigate down from pin button focus. Seems very fragile though 🤔

Reply:
Oh, I thought you were talking about #1249 , not #1178.  Right, good point.  I'll add it in.

Reply:
This is made a little complicated by how many ways there are to add and remove tiles.

Reply:
Hopefully gets cleaned up with https://github.com/mozilla-mobile/firefox-tv/issues/1085